### PR TITLE
Add Graql.parseLabel for label validation

### PR DESCRIPTION
**Pull request by alexjpwalker**
## What is the goal of this PR?

Because Graql type label validation is complex, we've introduced a method `Graql.parseLabel` that can be used to validate that a String is a valid type label.

## What are the changes implemented in this PR?

Add Graql.parseLabel

### CODE REVIEW

**Review by haikalpribadi — APPROVED** [review]

File: java/Graql.java [hunk 1]
```diff
@@ -81,6 +81,10 @@ public static BoundVariable parseVariable(String variable) {
         return parser.parseVariableEOF(variable);
     }
 
+    public static String parseLabel(String label) {
+        return parser.parseLabelEOF(label);
```
Comment (haikalpribadi) [inline java/Graql.java:85]: We should check here if the label contains whitespace or comments, and throw an exception if so.

**Review by haikalpribadi — CHANGES_REQUESTED** [review]

File: java/Graql.java [hunk 2]
```diff
@@ -81,6 +84,17 @@ public static BoundVariable parseVariable(String variable) {
         return parser.parseVariableEOF(variable);
     }
 
+    public static String parseLabel(String label) {
+        String parsedLabel;
+        try {
+            parsedLabel = parser.parseLabelEOF(label);
+        } catch (GraqlException e) {
+            throw GraqlException.of(ILLEGAL_CHAR_IN_LABEL.message(label));
```
Comment (alexjpwalker) [inline java/Graql.java:92]: Reproducible examples:
- `tx.concepts().putEntityType("1abc")`
- `tx.concepts().putEntityType("-_-")`

File: java/Graql.java [hunk 3]
```diff
@@ -81,6 +84,17 @@ public static BoundVariable parseVariable(String variable) {
...
+        }
+        if (!parsedLabel.equals(label)) throw GraqlException.of(PARSED_LABEL_DIFFERS_FROM_RAW.message(label, parsedLabel));
```
Comment (alexjpwalker) [inline java/Graql.java:94]: Reproducible examples:
- `tx.concepts().putEntityType(" abc")`
- `tx.concepts().putEntityType("person#123")`

File: java/common/exception/ErrorMessage.java [hunk 4]
```diff
@@ -32,69 +32,73 @@
     public static final ErrorMessage MISSING_DEFINABLES =
             new ErrorMessage(6, "The query has not been provided with any definables.");
     public static final ErrorMessage MATCH_HAS_NO_BOUNDING_NAMED_VARIABLE =
-            new ErrorMessage(37, "The match query does not have named variables to bound the nested disjunction/negation pattern(s).");
+            new ErrorMessage(7, "The match query does not have named variables to bound the nested disjunction/negation pattern(s).");
     public static final ErrorMessage MATCH_HAS_NO_NAMED_VARIABLE =
-            new ErrorMessage(7, "The match query has no named variables to retrieve.");
+            new ErrorMessage(8, "The match query has no named variables to retrieve.");
     public static final ErrorMessage MATCH_HAS_UNBOUNDED_NESTED_PATTERN =
-            new ErrorMessage(38, "The match query contains a nested pattern is not bounded: '%s'.");
```
Comment (alexjpwalker) [inline java/common/exception/ErrorMessage.java:39]: ??????????

File: java/Graql.java (hunk 3 above)
Comment (haikalpribadi) [reply java/Graql.java:94]: Let's just use one error message you have above.

**Review by haikalpribadi — CHANGES_REQUESTED** [review]

File: java/Graql.java [hunk 5]
```diff
@@ -81,6 +83,17 @@ public static BoundVariable parseVariable(String variable) {
...
     }
 
+    public static String parseLabel(String label) {
+        String parsedLabel;
+        try {
+            parsedLabel = parser.parseLabelEOF(label);
+        } catch (GraqlException e) {
+            throw GraqlException.of(ILLEGAL_CHAR_IN_LABEL.message(label));
+        }
+        if (!parsedLabel.equals(label)) throw GraqlException.of(ILLEGAL_CHAR_IN_LABEL.message(label)); // e.g: 'abc#123'
```
Comment (alexjpwalker) [inline java/Graql.java:93]: Resulting error message:
`[GQL39] Graql Error: 'person#123' is not a valid Type label. Type labels must start with a letter, and may contain only letters, numbers, '-' and '_'.`

**Review by haikalpribadi — APPROVED** [review]

**Pull request merged**